File: c/232. Implement Queue using Stacks.py
class MyQueue:
    def __init__(self):
        self.st1=[]
        self.st2=[]

    def push(self, x: int) -> None:
        self.st1.append(x)

    def pop(self) -> int:
        if len(self.st1)==0 and len(self.st2)==0:return -1
        
        if len(self.st2)==0:
            for _ in range(len(self.st1)):
                self.st2.append(self.st1.pop())
            
        return self.st2.pop()

    def peek(self) -> int:
        if len(self.st1)==0 and len(self.st2)==0:return -1

        if len(self.st2)==0:
            for _ in range(len(self.st1)):
                self.st2.append(self.st1.pop())
        
        return self.st2[-1]

    def empty(self) -> bool:
        return len(self.st1)==0 and len(self.st2)==0

q=MyQueue()
q.push(1)
q.push(3)
q.push(7)

print(q.peek())
print(q.empty())
print(q.pop())
print(q.peek())
print(q.pop())
print(q.peek())
print(q.pop())
print(q.peek())
print(q.empty())

File: c/232. Implement Queue using Stacks.js
var MyQueue = function() {
    this.st1=[];
    this.st2=[];
};

MyQueue.prototype.push = function(x) {
    this.st1.push(x);
};

MyQueue.prototype.pop = function() {
    if(this.st2.length===0 && this.st1.length===0)return -1;
    if(this.st2.length===0){
        while(this.st1.length>0){
            this.st2.push(this.st1.pop());
        }
    }
    return this.st2.pop();
};

MyQueue.prototype.peek = function() {
    if(this.st2.length===0){
        while(this.st1.length>0){
            this.st2.push(this.st1.pop());
        }
    }
    return this.st2.at(-1);
};

MyQueue.prototype.empty = function() {
    return this.st1.length===0 && this.st2.length===0;
};

let q=new MyQueue ();
q.push(1);
q.push(3);
q.push(7);

console.log(q.peek());
console.log(q.empty());
console.log(q.pop());
console.log(q.peek());
console.log(q);